### PR TITLE
fix: mf-2815 DSearch bugs

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-2815

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/ENS/src/SNSAdaptor/SearchResultInspector.tsx [hunk 1]
```diff
@@ -126,6 +126,15 @@ export const SearchResultInspectorContent = forwardRef(function (
     })
 
     const badgeCollectionRef = useRef<HTMLDivElement>(null)
+
+    useEffect(() => {
+        setIsShowSocialAccountList(
```
Comment (hanszhoou) [inline packages/plugins/ENS/src/SNSAdaptor/SearchResultInspector.tsx]: The mutation of `ref.current` is undetectable sometimes if rely on it directly. It's okay to set a state in `useEffect` if it is not a cycling.

Comment (guanbinrui) [inline packages/plugins/ENS/src/SNSAdaptor/SearchResultInspector.tsx]: Avoid using setState in `useEffect`.

File: packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx [hunk 2]
```diff
@@ -88,9 +89,25 @@ export function SocialAccountList({ nextIdBindings }: SocialAccountListProps) {
...
+    const onScroll = useCallback(() => {
+        setHideToolTip(true)
+        handleEndScroll()
+    }, [])
+
+    useEffect(() => {
+        const menu = ref.current?.querySelector('[role="menu"]')?.parentElement
+        menu?.addEventListener('scroll', onScroll)
+        return () => menu?.removeEventListener('scroll', onScroll)
+    }, [ref.current, JSON.stringify(nextIdBindings)])
```
Comment (guanbinrui) [inline packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx]: Add onScroll into deps.

File: packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx [hunk 3]
```diff
@@ -88,9 +89,25 @@ export function SocialAccountList({ nextIdBindings }: SocialAccountListProps) {
     const t = useI18N()
     const { classes, cx } = useStyles({ isMenuScroll: nextIdBindings.length > 5 })
     const position = useWindowScroll()
+    const [hideToolTip, setHideToolTip] = useState(false)
+    const ref = useRef<HTMLDivElement | null>(null)
+
+    const handleEndScroll = useMemo(() => debounce(() => setHideToolTip(false), 300), [])
+
+    const onScroll = useCallback(() => {
```
Comment (guanbinrui) [inline packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx:97]: Strange to me. Why not make the callback a debounce one.

Comment (zhouhanseng) [reply packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx:97]: You mean `const handleEndScroll = useCallback(() => debounce(() => setHideToolTip(false), 300), [])`, I tried it, but `useMemo` works.

Comment (UncleBill) [reply packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx:97]: Should pass the debounced function to `useCallback` instead
```ts
const handleEndScroll = useCallback(debounce(() => setHideToolTip(false), 300), [])
```

Comment (UncleBill) [reply packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx:97]: `useCallback` is somehow a grammar sugar of `useMemo` of function

Comment (hanszhoou) [reply packages/plugins/ENS/src/SNSAdaptor/SocialAccountList.tsx:97]: I tried, 
`const handleEndScroll = useCallback(debounce(() => setHideToolTip(false), 300), [])` does not work.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**